src/modules/ticket-fields/fields/LookupField.tsx

```diff
@@ -158,14 +173,29 @@ export function LookupField({
           `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
         );
 
-        const data = await response.json();
+        const data: AssetsApiResponse = await response.json();
         if (response.ok) {
-          let fetchedOptions = data.custom_object_records.map(
-            ({ name, id }: { name: string; id: string }) => ({
-              name,
-              value: id,
-            })
-          );
+          let fetchedOptions: TicketFieldOptionObject[] =
+            data.custom_object_records.map((item) => {
+              const isAsset = item.custom_object_key === "standard::itam_asset";
+
+              return {
+                name: item.name,
+                value: item.id,
+                item_asset_type_id: isAsset
+                  ? (item.custom_object_fields as AssignedAssetFields)[
+                      "standard::asset_type"
+                    ] ?? ""
+                  : "",
```

Comment:
I think we should be able to tweak `applyAssetFiltersAsync` so this logic doesn't need to be added in this component.